src/argilla/client/feedback/schemas/metadata.py

```diff
@@ -161,21 +164,33 @@ def _all_values_exist(self, introduced_value: Optional[Union[str, List[str]]] =
 
         return introduced_value
 
+    def _check_allowed_value_type(self, value: Any) -> Any:
+        if value is not None:
+            if isinstance(value, str):
+                return value
+            if isinstance(value, list):
+                return [self._check_allowed_value_type(v) for v in value]
+            raise TypeError(
+                f"Provided '{self.name}={value}' of type {type(value)} is not valid, "
+                "only values of type `str` or `list` are allowed."
+            )
+        return value
```

Comment:
A possible simplification/refactor:

```suggestion
    def _check_allowed_value_type(self, value: Any) -> Any:
        if value is None or isinstance(value, str):
            return value
        
        if isinstance(value, list):
            return [self._check_allowed_value_type(v) for v in value]
        
        raise TypeError(
            f"Provided '{self.name}={value}' of type {type(value)} is not valid, "
            "only values of type `str` or `list` are allowed."
        )
```